src/muse/mca.py

```diff
@@ -243,38 +243,14 @@ def update_carbon_budget(self, market: Dataset, year_idx: int) -> float:
         )
 
     def update_carbon_price(self, market) -> float | None:
-        """Calculates the updated carbon price, if required.
-
-        If the emission calculated for the next time period is larger than the
-        limit, then the carbon price needs to be updated to ensure that whatever the
-        sectors do, the carbon budget limit is not exceeded.
+        """Calculates the updated carbon price.
 
         Arguments:
-            market: Market, with the prices, supply, consumption and demand.
+            market: Market with the prices, supply, consumption and demand.
 
         Returns:
-            The new carbon price or None
+            The new carbon price. If None, the price is not updated.
```

Comment:
I don't think the carbon price can be `None` anymore. If it can, then the docstrings for the different carbon methods and their signature needs updating.

Reply:
I've addressed this in #484